shell.nim

```diff
@@ -197,7 +227,15 @@ proc concatCmds(cmds: seq[string], sep = " && "): string =
   ## concat commands to single string, by default via `&&`
   result = cmds.join(sep)
 
-proc asgnShell*(cmd: string): tuple[output: string, exitCode: int] =
+
+# NOTE Checking for key in set, calling function on each `asgnShell`
+# invokation might decrease performance. Although not printing
+# anything into stdout might increase performace.
```

Comment:
I'm pretty sure that compared to the overhead of invoking a shell in the first place a simple check of "flag in set" is negligible.